model/policy.go

```diff
@@ -247,7 +247,12 @@ func (model Model) RemovePolicy(sec string, ptype string, rule []string) bool {
 		return false
 	}
 
-	model[sec][ptype].Policy = append(model[sec][ptype].Policy[:index], model[sec][ptype].Policy[index+1:]...)
+	if index < len(model[sec][ptype].Policy)-1 {
```

Comment:
What does this code mean?

Reply:
Ensure that the index exists in the slice, and using copy is more efficient than append and can reduce memory allocation.

Reply:
@hsluoyz

Reply:
<img width="1003" alt="image" src="https://github.com/casbin/casbin/assets/22953612/7b1ab68d-ad95-4239-95dd-26634192309e">

Reply:
Use append instead of copy. Readability is most important